### PR TITLE
feat(formula): updates and fixes

### PR DESCRIPTION
**Pull request by DianaLease**
# Closes #<CORRESPONDING ISSUE NUMBER>

### Changes
- refactor(Formula): refactor code and fix so that dependencies are not hardcoded
- fix(formula): fix hovering style so it does not overlap actual formula and shows pointer triangle
- fix(formula): fixes issue where clicking formula multiple times in safari causes crash
- feat(formula): change style to have different font color with no border

### CODE REVIEW

**Comment by Michael-Grover** [comment]
Please remove purple border and white background around formula and change formula text color to #AE00BE. This it to make it more clear that formulae can not be directly edited like variables can.

image

File: packages/ui-contract-editor/src/lib/ContractEditor/index.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ const contractProps = {
  */
 const ContractEditor = (props) => {
   const [hoveringFormulaContract, setHoveringFormulaContract] = useState(false);
-  const [formulaDependents, setFormulaDependents] = useState({});
+  const [formulaNode, setFormulaNode] = useState(null);
```
Comment (jeromesimeon) [inline packages/ui-contract-editor/src/lib/ContractEditor/index.js:81]: This `null` thing confuses me a little bit. Why did it change?

Comment (DianaLease) [reply packages/ui-contract-editor/src/lib/ContractEditor/index.js:81]: This used to be an object that we would set `node` and `dependencies` properties on. Now the node itself has `dependencies` on its data property, so we only need to keep the formula node on state. Initiating it as `null` lets me do the null check on line 90 to ensure we don't try highlighting variables if we don't have a formula node.

**Comment by DianaLease** [comment]
@Michael-Grover This has been updated to remove the border and change the font color for formulas

File: packages/ui-contract-editor/src/lib/ContractEditor/index.js (hunk 1 above)
Comment (jeromesimeon) [reply packages/ui-contract-editor/src/lib/ContractEditor/index.js:81]: very nice. thanks for explaining.

**Pull request merged**